libnetwork/drivers/bridge/bridge.go

```diff
@@ -685,8 +690,8 @@ func (d *driver) createNetwork(config *networkConfiguration) (err error) {
 		id:           config.ID,
 		endpoints:    make(map[string]*bridgeEndpoint),
 		config:       config,
-		portMapper:   portmapper.New(d.config.UserlandProxyPath),
-		portMapperV6: portmapper.New(d.config.UserlandProxyPath),
+		portMapper:   portmapper.NewWithPortAllocator(d.portAllocator, d.config.UserlandProxyPath),
+		portMapperV6: portmapper.NewWithPortAllocator(d.portAllocator, d.config.UserlandProxyPath),
```

Comment:
Interesting; looks like this function was added in https://github.com/moby/libnetwork/commit/9da342a7fc91c68a46e0a9dd746b80359d92467a, but (at a glance) never used, other than to back `New()` 🤔 🤷‍♂️